scripts/g.extension/g.extension.py

```diff
@@ -2218,7 +2234,9 @@ def resolve_known_host_service(url, name, branch):
         else:
             actual_start = ""
         if "branch" in match["url_end"]:
-            suffix = match["url_end"].format(name=name, branch=branch)
+            suffix = match["url_end"].format(
+                name=name, branch=branch if branch else "main"
```

Comment:
Perhaps a comment here about the assumptions. This resolves to a zip, so branch name is needed and there is no easy default. Getting default branch name for different services would be more work, so here I would say "main" is close enough, esp. since it is unrelated to the goal of this PR.

Reply:
Yes, see also comment above.